Service/WorkoutService.cs

```diff
@@ -44,11 +39,17 @@ public async Task<IEnumerable<Workout>> GetAllAsync(CancellationToken cancellati
         {
             if (id <= 0)
```

Comment:
```suggestion
            if (id <= 0)
```
Проверка ни о чем не говорит. Кроме как о том, что злоумышленник знает что айдишник должен быть больше 0)) В принципе если мы знаем, что таких идентификаторов у нас быть не может и никогда не будет, то проще возвращать null. Ну то есть тренировка с таким айдишником не найдена.
Пусть будет лучше return null; в этом месте